searchcore/src/tests/proton/server/memory_flush_config_updater/memory_flush_config_updater_test.cpp

```diff
@@ -75,6 +75,55 @@ TEST_F("require that strategy is updated when setting new config", Fixture)
     TEST_DO(f.assertStrategyConfig(6, 3, 30));
 }
 
+void
+expectEqual(const MemoryFlush::Config & a, const MemoryFlush::Config & b) {
+    EXPECT_TRUE(a.equal(b));
+    EXPECT_TRUE(a == b);
+    EXPECT_FALSE( a != b);
+    EXPECT_TRUE(b.equal(a));
+    EXPECT_TRUE(b == a);
+    EXPECT_FALSE( b != a);
+}
+
+void
+expectNotEqual(const MemoryFlush::Config & a, const MemoryFlush::Config & b) {
+    EXPECT_FALSE(a.equal(b));
+    EXPECT_FALSE(a == b);
+    EXPECT_TRUE( a != b);
+    EXPECT_FALSE(b.equal(a));
+    EXPECT_FALSE(b == a);
+    EXPECT_TRUE( b != a);
+}
+
+TEST("require that MemoryFlush::Config equal is correct") {
+    MemoryFlush::Config a, b;
+    expectEqual(a, b);;
```

Comment:
Nitpick: Some extra ";" have been added. Applies further down as well.

Reply:
Fixed in #18387